src/iproject.ts

```diff
@@ -68,39 +106,127 @@ export class IProject {
     return Uri.file(path.join(this.workspaceFolder.uri.fsPath, `.env`));
   }
 
-  public async read() {
-    const content = await workspace.fs.readFile(this.getIProjFilePath());
-    this.state = IProject.validateIProject(content.toString());
+  public async addToIncludePaths(includePath: string) {
+    const unresolvedState = await this.getUnresolvedState();
+
+    if (unresolvedState) {
+      if (unresolvedState.includePath) {
+        if (!unresolvedState.includePath.includes(includePath)) {
+          unresolvedState.includePath.push(includePath);
+        } else {
+          window.showErrorMessage(`${includePath} already exists in includePaths`);
+          return;
+        }
+      } else {
+        unresolvedState.includePath = [includePath];
+      }
+
+      await this.updateIProj(unresolvedState);
+    } else {
+      window.showErrorMessage('No iproj.json found');
+    }
   }
 
-  public async addToIncludePaths(includePath: string) {
-    const iProjExists = await this.projectFileExists('iproj.json');
-    if (iProjExists) {
-      const content = await workspace.fs.readFile(this.getIProjFilePath());
+  public async addToLibraryList(library: string, position: 'preUsrlibl' | 'postUsrlibl') {
+    const unresolvedState = await this.getUnresolvedState();
+    const state = await this.getState();
+
+    if (unresolvedState && state) {
+      if (unresolvedState[position] && state[position]) {
+        if (state[position]!.includes(library)) {
+          window.showErrorMessage(`${library} already exists in ${position}`);
+          return;
 
-      const iProject = IProject.validateIProject(content.toString());
-      if (iProject) {
-        try {
-          if (iProject.includePath) {
-            if (!iProject.includePath.includes(includePath)) {
-              iProject.includePath.push(includePath);
-            } else {
-              window.showErrorMessage(`${includePath} already exists in includePaths`);
+        } else {
+          unresolvedState[position]!.push(library);
+        }
+      } else {
+        unresolvedState[position] = [library];
+      }
+
+      await this.updateIProj(unresolvedState);
+    } else {
+      window.showErrorMessage('No iproj.json found');
+    }
+  }
+
+  public async setCurrentLibrary(library: string) {
+    const unresolvedState = await this.getUnresolvedState();
+    const state = await this.getState();
+
+    if (unresolvedState && state) {
+      if (state.curlib === library) {
+        window.showErrorMessage(`Current library already set to ${library}`);
+        return;
+      } else if (unresolvedState.curlib && unresolvedState.curlib.startsWith('&')) {
+        //Update variable
+        await this.setEnv(unresolvedState.curlib.substring(1), library);
+      } else {
+        await this.setEnv(DEFAULT_CURLIB, library);
+
+        unresolvedState.curlib = DEFAULT_CURLIB;
+      }
+
```

Comment:
should there be a side effect on the resolved value of iProj.objlib if it was defaulting to curlib, or do we just let BOB resolve that at build time?

Reply:
Pushed a change to default the `objlib` to `curlib` if it is not defined regardless if it is a resolved variable or a hardcoded library.